model/lifecycle.go

```diff
@@ -536,9 +536,11 @@ func addTasksToBuild(ctx context.Context, creationInfo TaskCreationInfo) (*build
 		if !utility.StringSliceContains(tasksWithActivationTime, t.DisplayName) {
 			continue
 		}
-		activateTaskAt, err := creationInfo.ProjectRef.GetActivationTimeForTask(
-			creationInfo.Project.FindTaskForVariant(t.DisplayName, creationInfo.Build.BuildVariant),
-			t.Id)
+		bvtu := creationInfo.Project.FindTaskForVariant(t.DisplayName, creationInfo.Build.BuildVariant)
+		if !bvtu.HasSpecificActivation() {
```

Comment:
I always have a bit of difficulty understanding how specific activation works when defined in the BV vs BVTU, since the settings precedence doesn't work the same as the other task fields - Does this also handle the case when you set the specific activation condition at the BV level rather than the BVTU level? I thought that specific activation fields like cron/batchtime weren't propagated down to the BVTU level and we had to [check the BV settings explicitly](https://github.com/evergreen-ci/evergreen/blob/332598845126ee2c235813a87661043598c880ab/model/generate.go#L538) before [checking the BVTU settings](https://github.com/evergreen-ci/evergreen/blob/332598845126ee2c235813a87661043598c880ab/model/generate.go#L551).

Reply:
This only checks the build variant task level but that's the correct thing to check for getting the task activation time -- if it's set for the variant, that'll be part of the variant's activation info (this allows us to set one activation for the variant, and a different one for individual tasks that are overriding it).